File: backend/animedix/src/main/java/com/psleziona/animedix/model/AnimalSurgery.java
package com.psleziona.animedix.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Entity
@Data
@NoArgsConstructor(force = true)
public class AnimalSurgery {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idAnimalSurgery;
    @ManyToOne
    @JoinColumn(name = "id_animal")
    @JsonIgnoreProperties({"owner","visits"})
    private Animal animal;
    @ManyToOne
    @JoinColumn(name = "id")
    @JsonIgnoreProperties({"shifts","surgeries","visits"})
    private Employee doctor;
    @ManyToOne
    @JoinColumn(name = "id_surgery")
    private Surgery surgery;
    @NonNull
    private LocalDateTime date;
    @OneToMany(mappedBy = "animalSurgery")
    private List<UsedAssortment> usedAssortment;
}

File: frontend/src/app/_model/animal-surgery.ts
import {Animal} from "./animal";
import {Employee} from "./employee";
import {Surgery} from "./surgery";
import {Assortment} from "./assortment";
import {UsedAssortment} from "./used-assortment";

export interface AnimalSurgery {
  idAnimalSurgery?: number,
  animal?: Animal,
  doctor?: Employee,
  surgery?: Surgery,
  date?: string,
  usedAssortment?: UsedAssortment[]
}
